packages/api/src/controllers/helpers.ts

```diff
@@ -771,45 +771,49 @@ export async function addDefaultProjectId(
     }
   };
 
-  const enrichResponseWithUserProjectId = (document) => {
+  const enrichResponseWithUserProjectId = async (document) => {
     if ("id" in document && "userId" in document && "user" in document) {
-      if (
-        (!document.projectId || document.projectId === "") &&
-        document.user?.defaultProjectId
-      ) {
-        document.projectId = document.user.defaultProjectId;
+      if (!document.projectId || document.projectId === "") {
+        if (document.user?.defaultProjectId) {
+          document.projectId = document.user.defaultProjectId;
+        } else {
+          let user = await db.user.get(document.userId, { useCache: true });
```

Comment:
We shouldn't be fetching the user if they are already the same as the caller.
```suggestion
          let user = document.userId === req.user.id ? req.user : await db.user.get(document.userId, { useCache: true });
```
(good call on caching)